Source/WebKit/GPUProcess/graphics/WebGPU/RemoteBindGroup.cpp

```diff
@@ -44,7 +44,16 @@ RemoteBindGroup::RemoteBindGroup(PAL::WebGPU::BindGroup& bindGroup, WebGPU::Obje
     m_streamConnection->startReceivingMessages(*this, Messages::RemoteBindGroup::messageReceiverName(), m_identifier.toUInt64());
 }
 
-RemoteBindGroup::~RemoteBindGroup() = default;
+RemoteBindGroup::~RemoteBindGroup()
+{
+    destroy();
```

Comment:
I'm super confused about this. When we receive the Destroy message, we run destroy(), which removes us from the object heap. That will cause our refcount to reach 0, which will cause us to be destroyed, which will run our destructor.... which will run the destroy() function again???

Why isn't it just:
```
RemoteBindGroup::~RemoteBindGroup() = default;

void RemoteBindGroup::destroy()
{
    m_objectHeap.removeObject(m_identifier);
    // we're dead now, don't access anything! return immediately.
}
```
and that's it?

Reply:
I think that's right, we shouldn't need two calls to destroy. I'll try with the default destructor approach